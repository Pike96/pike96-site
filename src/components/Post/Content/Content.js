// @flow strict
import React from 'react';
import styles from './Content.module.scss';
import Meta from '../Meta';
import Tags from '../Tags';
import type { Fields, FrontMatter } from '../../../types';

type Props = {
  body: string,
  title: string,
  fields: Fields,
  frontmatter: FrontMatter
};

const Content = ({
  body, title, fields, frontmatter
}: Props) => {
  const { tagSlugs } = fields;
  const { tags, date } = frontmatter;

  return (
    <div className={styles['content']}>
      <h1 className={styles['content__title']}>{title}</h1>
      {tags && tagSlugs && <Tags tags={tags} tagSlugs={tagSlugs} />}
      <Meta date={date} />
      <div className={styles['content__body']} dangerouslySetInnerHTML={{ __html: body }} />
    </div>
  );
};

export default Content;
