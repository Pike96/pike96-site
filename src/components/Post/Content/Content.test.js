// @flow strict
import React from 'react';
import renderer from 'react-test-renderer';
import Content from './Content';

describe('Content', () => {
  it('renders correctly', () => {
    const props = {
      body: '<p>test</p>',
      title: 'test',
      fields: {
        slug: '/test',
        categorySlug: '/test-category',
        tagSlugs: [
          '/test_0',
          '/test_1'
        ]
      },
      frontmatter: {
        date: '2016-09-01',
        tags: [
          'test_0',
          'test_1'
        ],
        title: 'test'
      },
    };

    const tree = renderer.create(<Content {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
