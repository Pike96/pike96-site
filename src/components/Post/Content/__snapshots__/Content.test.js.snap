// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content renders correctly 1`] = `
<div
  className="content"
>
  <h1
    className="content__title"
  >
    test
  </h1>
  <div
    className="tags"
  >
    <ul
      className="tags__list"
    >
      <li
        className="tags__list-item"
      >
        <a
          className="tags__list-item-link"
          href="/test_0"
        >
          test_0
        </a>
      </li>
      <li
        className="tags__list-item"
      >
        <a
          className="tags__list-item-link"
          href="/test_1"
        >
          test_1
        </a>
      </li>
    </ul>
  </div>
  <div
    className="meta"
  >
    <p
      className="meta__date"
    >
      1 Sep 2016
    </p>
  </div>
  <div
    className="content__body"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p>test</p>",
      }
    }
  />
</div>
`;
